src/ttt/game_rules.clj

```diff
@@ -0,0 +1,57 @@
+(ns ttt.game-rules
+  (:require [clojure.math.numeric-tower :as math]))
+
+(defn- get-board-size
+  [board]
+  (math/sqrt (count board)))
+
+(defn- create-columns
+  [board]
+ (apply mapv vector (partition (get-board-size board) board)))
+
+(defn- create-rows
+  [board]
+  (apply mapv vector (create-columns board)))
+  
+(defn- get-cell-value
+  [row-with-index]
+  (first (map (fn [[index row]] (row index)) row-with-index)))
+
+(defn- create-diagonal
+  [cells]
+    (let [row (map-indexed hash-map cells)]
+      (into [] (map get-cell-value row))))
+
+(defn- create-left-to-right-diagonal
+  [board]
+  (let [rows (create-rows board)]
+    (create-diagonal rows)))
+
+(defn- create-right-to-left-diagonal
+  [board]
+  (let [rows (reverse (create-rows board))]
+    (create-diagonal rows)))
+
+(defn- create-groups-of-cells
+  [board]
+  [(create-rows board) (create-columns board) [(create-left-to-right-diagonal board) (create-right-to-left-diagonal board)]])
+
+(defn- check-group-of-cells
+  [cells]
+  (map (fn[collection] (= 1 (count (set collection)))) cells))
+
+(defn- available-spaces?
+  [board]
+  (boolean (some true? (map number? board))))
```

Comment:
I believe you can just do (some number? board) for the same thing. Also, in other cases, I believe you can use `some?` instead of doing `some true?`

Reply:
one more good thing is that it seems like `some?` returns a boolean false for the falsey case

Reply:
acttually sorry, I tried it on repl and it's not what I thought. It's just `not-nil`